automation_tools/repository.py

```diff
@@ -0,0 +1,160 @@
+    --disable``.
+
+    For example::
+
+        disable_repos('repo1', 'repo2')
+
+    Will run the command ``subscription-manager repos --disable repo1 repo2``.
+
+
+    """
+    run('subscription-manager repos --disable {0}'
+        .format(' '.join(['"{0}"'.format(repo) for repo in args])))
+
+
+def delete_custom_repos(**args):
+    """Delete repos files on ``/etc/yum.repos.d``.
+
+    All files that match ``<filename>.repo`` will be deleted, double check when
+    running this task because it can delete repository files created by
+    ``subscription-manager`` and other tools.
+
+    :raise: ``ValueError`` if repository name is 'redhat'.
+
+    """
+    for name in args:
+        name = name.rstrip('.repo')
+        if name == 'redhat':
+            raise ValueError('This task will not delete redhat.repo file.')
+        run('rm -f /etc/yum.repos.d/{0}.repo'.format(name), warn_only=True)
+
+
+def enable_repos(*args, **kwargs):
+    """Enable repos passed as ``args`` using ``subscription-manager repos
+    --enable``.
+
+    For example::
+
+        enable_repos('repo1', 'repo2')
+
+    Will run the command ``subscription-manager repos --enable repo1 repo2``.
+
+    """
+    run('subscription-manager repos --enable {0}'
+        .format(' '.join(['"{0}"'.format(repo) for repo in args])))
+
+
+def create_custom_repos(**kwargs):
+    """Create custom repofiles.
+
+    Each ``kwargs`` item will result in one repository file created. Where the
+    key is the repository filename and repository name, and the value is the
+    repository URL.
+
+    For example::
+
+        create_custom_repo(custom_repo='http://repourl.domain.com/path')
+
+    Will create a repository file named ``custom_repo.repo`` with the following
+    contents::
+
+        [custom_repo]
+        name=custom_repo
+        baseurl=http://repourl.domain.com/path
+        enabled=1
+        gpgcheck=0
+
+    """
+    for name, url in kwargs.items():
+        repo_file = StringIO()
+        repo_file.write(
+            '[{name}]\n'
+            'name={name}\n'
+            'baseurl={url}\n'
+            'enabled=1\n'
+            'gpgcheck=0\n'
+            .format(name=name, url=url)
+        )
+        put(local_path=repo_file,
+            remote_path='/etc/yum.repos.d/{0}.repo'.format(name))
+        repo_file.close()
+
+
+def enable_satellite_repos(cdn=False, disable_enabled=True):
+    """Enable repositories required to install Satellite 6
+
+    If ``disable_enabled`` is ``True``, then this task will first disable any
+    already enabled repository, including beaker ones.
+
+    """
+    if isinstance(cdn, str):
+        cdn = (cdn.lower() == 'true')
+    if isinstance(disable_enabled, str):
+        disable_enabled = (disable_enabled.lower() == 'true')
+
+    if disable_enabled is True:
+        disable_beaker_repos()
+        disable_repos('*')
+
+    # Disable yum plugin for sub-man
+    run('sed -i -e "s/^enabled.*/enabled=0/" '
```

Comment:
We can improve this by searching for enabled equals 1. And the file will be edited only if enabled = 1